python/strings_udf/cpp/src/strings/udf/udf_apis.cu

```diff
@@ -36,13 +76,47 @@ std::unique_ptr<rmm::device_buffer> to_string_view_array(cudf::column_view const
                 .release()));
 }
 
+/**
+ * @copydoc column_from_udf_string_array
+
+ * @param stream CUDA stream used for allocating/copying device memory and launching kernels
+ */
+std::unique_ptr<cudf::column> column_from_udf_string_array(udf_string* d_strings,
+                                                           cudf::size_type size,
+                                                           rmm::cuda_stream_view stream)
+{
+  // create string_views of the udf_strings
+  auto indices = rmm::device_uvector<cudf::string_view>(size, stream);
+  thrust::transform(rmm::exec_policy(stream),
+                    d_strings,
+                    d_strings + size,
+                    indices.data(),
+                    udf_string_to_string_view_transform_fn{});
+
+  auto results = cudf::make_strings_column(indices, cudf::string_view(nullptr, 0), stream);
```

Comment:
This requires two allocations, one for the `device_uvector` and then a second for `make_strings_column`. Is there any way to avoid the first one, or would the resulting code be too verbose since working with elements of a string column is less ergonomic than operations on a `cudf::string_view`? You could conceivably have an operator that first creates a `string_view` and then assigns to each element of the column, for instance, but I don't know how painful that would be offhand.

Reply:
I think this is unavoidable. The biggest part of this is `make_strings_column` which is alot of code written specifically to gather individual strings from device memory into a strings column. The extra allocate and conversion here would not make that code any faster.